src/core/src/service_interfaces/TelemetryWriter.py

```diff
@@ -14,131 +14,123 @@
+            bytes_dropped = len(formatted_message) - message_size_limit_in_bytes + Constants.TELEMETRY_BUFFER_FOR_DROPPED_COUNT_MSG_IN_BYTES
+            return formatted_message[:message_size_limit_in_bytes - Constants.TELEMETRY_BUFFER_FOR_DROPPED_COUNT_MSG_IN_BYTES].decode('utf-8') + '. [{0} bytes dropped]'.format(bytes_dropped)
+
+        return formatted_message
+
+    def write_event(self, task_name, message, event_level=Constants.TelemetryEventLevel.Informational):
+        """ Creates and writes event to event file after validating none of the telemetry size restrictions are breached """
+        self.__delete_older_events()
+
+        event = self.__new_event_json(task_name, event_level, message)
+        if len(json.dumps(event)) > Constants.TELEMETRY_EVENT_SIZE_LIMIT_IN_BYTES:
+            print("Cannot send data to telemetry as it exceeded the acceptable data size. [Data not sent={0}]".format(json.dumps(message)))
+        else:
+            self.write_event_using_temp_file(self.events_folder_path, event)
+
+    def __delete_older_events(self):
+        """ Delete older events until the at least one new event file can be added as per the size restrictions """
+        if self.__get_events_dir_size() < Constants.TELEMETRY_DIR_SIZE_LIMIT_IN_BYTES - Constants.TELEMETRY_EVENT_FILE_SIZE_LIMIT_IN_BYTES:
+            # Not deleting any existing event files as the event directory does not exceed max limit. At least one new event file can be added. Not printing this statement as it will add repetitive logs
+            return
+
+        print("Events directory size exceeds maximum limit. Deleting older event files until at least one new event file can be added.")
+        event_files = [os.path.join(self.events_folder_path, event_file) for event_file in os.listdir(self.events_folder_path) if (event_file.lower().endswith(".json"))]
+        event_files.sort(key=os.path.getmtime, reverse=True)
+
+        for event_file in event_files:
+            try:
+                if self.__get_events_dir_size() < Constants.TELEMETRY_DIR_SIZE_LIMIT_IN_BYTES - Constants.TELEMETRY_EVENT_FILE_SIZE_LIMIT_IN_BYTES:
+                    # Not deleting any more event files as the event directory has sufficient space to add at least one new event file.  Not printing this statement as it will add repetitive logs
+                    break
+
+                if os.path.exists(event_file):
+                    os.remove(event_file)
+                    print("Deleted event file. [File={0}]".format(repr(event_file)))
+            except Exception as e:
+                print("Error deleting event file. [File={0}] [Exception={1}]".format(repr(event_file), repr(e)))
+
+        if self.__get_events_dir_size() >= Constants.TELEMETRY_DIR_SIZE_LIMIT_IN_BYTES:
+            raise Exception("Older event files were not deleted. Current event will not be sent to telemetry as events directory size exceeds maximum limit")
+
```

Comment:
Please introduce a telemetry writer only log level in composite logger (that doesn't loop back to telemetry) - all of the code here (and below) are running blind aside from exceptions.

Important: refrain from using print in general but especially inside telemetry logger. The file mirror catch is to only capture things we have inadvertently missed or it's early in the bootstrap process.